pkg/userdata/coreos/provider.go

```diff
@@ -112,13 +113,12 @@ func (p Provider) UserData(
 	if err != nil {
 		return "", fmt.Errorf("failed to execute user-data template: %v", err)
 	}
-
-	return b.String(), nil
+	clean := regexp.MustCompile(`(?m)^[ \t]+$`).ReplaceAllString(b.String(), "")
```

Comment:
Same as https://github.com/kubermatic/machine-controller/pull/513/files#r280372518

On top: If that is necessary & we use it in multiple places, should we extract it into a dedicated function to avoid the duplication?

Reply:
It's only a one-liner. But yes, the function name may be better than a comment and also the function comment in implementation can be more descriptive.